### PR TITLE
Added Google Authentication

### LINKED ISSUE #30: Create authentication service

### PR DESCRIPTION
**Pull request by Tamas-Leung**
Added simple login page, login redirect and saving of token in local storage. Works with signing into google

Fixes #30

### CODE REVIEW

**Review by dipesube — APPROVED** [review]

File: src/frontend/src/app/login/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  ) {}
+
+  ngOnInit(): void {
+    // @ts-ignore
+    window.onGoogleLibraryLoad = () => {
+      // @ts-ignore
+      google.accounts.id.initialize({
+        // Ref: https://developers.google.com/identity/gsi/web/reference/js-reference#IdConfiguration
+        client_id:
+          '1049277800628-djvnt32n0odf07kqelkn1t0iee33kk0n.apps.googleusercontent.com',
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/login/login.component.ts:25]: Make this an environment variable

File: src/frontend/src/app/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        callback: this.handleCredentialResponse.bind(this), // Whatever function you want to trigger...
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/login/login.component.ts:26]: Remove comment

File: src/frontend/src/app/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        auto_select: false,
+        cancel_on_tap_outside: false,
+      });
+
+      // @ts-ignore
+      // google.accounts.id.prompt((notification: PromptMomentNotification) => {});
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/login/login.component.ts:32]: Do we need this?

File: src/frontend/src/app/services/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor(private httpClient: HttpClient, private router: Router) {}
+
+  logout = () => {
+    localStorage.removeItem('token');
+    console.log('token deleted');
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/services/auth/auth.service.ts:14]: Remove console.log

File: src/frontend/src/app/services/auth/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    // this.router.navigateByUrl('/login', { skipLocationChange: false });
```
Comment (YoussefRizkalla) [inline src/frontend/src/app/services/auth/auth.service.ts:16]: Do we need this?

File: src/backend/src/routes/auth.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const router = Router();
+
+/**
+ * @swagger
+ * /auth/login:
+ *   post:
+ *     tags:
+ *        - auth
+ *     summary: Login into service
+ *     description: Adds user to database if doesnt exist
```
Comment (YoussefRizkalla) [inline src/backend/src/routes/auth.js:12]: nit: if it doesn't exist.

**Review by CSchank — COMMENTED** [review]
Some late night OAuth'ing I see 😉

**Review by AntonKanug — APPROVED** [review]

**Pull request merged**